lib/spack/env/cc

```diff
@@ -175,15 +175,26 @@ while [ -n "$1" ]; do
             ;;
         -Wl,*)
             arg="${1#-Wl,}"
-            if [ -z "$arg" ]; then shift; arg="$1"; fi
-            if [[ "$arg" = -rpath=* ]]; then
-                rpaths+=("${arg#-rpath=}")
-            elif [[ "$arg" = -rpath ]]; then
+            # TODO: Handle multiple -Wl, continuations of -Wl,-rpath
+            if [[ $arg == '-rpath='* ]]; then
```

Comment:
This uses Bash's `[[` operator, which has simpler quoting rules than the `[` command. Essentially, within `[[`, shell variables are expanded, but are not split at white space. I prefer `[[` over `[` for this reason.

I'm quoting `-rpath` because it begins with a dash, which could otherwise be interpreted as a command (`-r` checks whether a file is readable).